lib/pkcs11/pkcs11_config.c

```diff
@@ -420,16 +420,52 @@ static CK_RV pkcs11_config_parse_freeslots(pkcs11_slot_ctx_ptr slot_ctx, char* c
     return CKR_OK;
 }
 
+static void pkcs11_config_parse_id(pkcs11_object_ptr pObject, char * argv[], int arg_c, int arg_id)
+{
+#if PKCS11_ID_SIZE > 0
+    // ID is hex number, of any length
+    // it is used to fill an id byte array of fixed length
+    char * id = "";
+    size_t id_end = 0;
+    size_t id_start = 0;
+    int i;
+
+    if (arg_c > arg_id)
+    {
+        id = argv[arg_id];
+        id_end = strlen(id);
+    }
+    for (int i = PKCS11_ID_SIZE-1; i >= 0; i--)
+    {
+        if (id_end <= 2) {
+            id_start = 0;
+        } else {
+            id_start = id_end - 2;
+        }
+        if (id_start ==  id_end)
+        {
+            pObject->id[i] = 0;
+        }
+        else
+        {
+            pObject->id[i] = (uint8_t)strtoul(id + id_start, NULL, 16);
+            id_end = id_start;
+            id[id_end] = '\0';
+        }
+    }
+#endif
+}
+
 static CK_RV pkcs11_config_parse_object(pkcs11_slot_ctx_ptr slot_ctx, char* cfgstr)
 {
-    char * argv[5];
-    int argc = (int)sizeof(argv);
```

Comment:
Note, fixed bug here, as argc = (int)sizeof(argv) sets 6*sizeof(char*) and not 6.